src/settings/SettingsPage.tsx

```diff
@@ -9,62 +9,67 @@ import {
     Stack,
     Typography,
 } from '@mui/material';
+import { useMutation } from '@tanstack/react-query';
 import { useState } from 'react';
 import {
     Form,
     Labeled,
     TextField,
     TextInput,
+    useDataProvider,
     useGetIdentity,
     useGetOne,
     useNotify,
     useUpdate,
 } from 'react-admin';
 import { useFormState } from 'react-hook-form';
 import { USER_STORAGE_KEY } from '../authProvider';
-import { UpdatePassword } from './UpdatePassword';
 import ImageEditorField from '../misc/ImageEditorField';
+import { UpdatePassword } from './UpdatePassword';
+import { SalesFormData } from '../types';
+import { CustomDataProvider } from '../dataProvider';
 
 export const SettingsPage = () => {
-    const [update] = useUpdate();
     const [isEditMode, setEditMode] = useState(false);
-    const { identity, refetch } = useGetIdentity();
-    const user = useGetOne('sales', { id: identity?.id });
+    const { identity, refetch: refetchIdentity } = useGetIdentity();
+    const { data, refetch: refetchUser } = useGetOne('sales', {
+        id: identity?.id,
+    });
     const notify = useNotify();
+    const dataProvider = useDataProvider<CustomDataProvider>();
+
+    const { mutate } = useMutation({
+        mutationKey: ['signup'],
+        mutationFn: async (data: SalesFormData) => {
+            if (!identity) {
+                throw new Error('Record not found');
+            }
+            return dataProvider.salesUpdate(identity.id, data);
+        },
+        onSuccess: data => {
+            // Update local user
+            localStorage.setItem(USER_STORAGE_KEY, JSON.stringify(data));
```

Comment:
**question:** Is local storage still required ? As we do not use it we supabase ?